src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResource.kt

```diff
@@ -40,29 +44,120 @@ class VisitResource(
         responseCode = "400",
         description = "Incorrect request to Get visits for prisoner",
         content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions retrieve a visit",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Visit not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+    ]
+  )
+  fun getVisitById(
+    @Schema(description = "visit id", example = "45645", required = true)
+    @PathVariable visitId: Long
+  ): VisitDto =
+    visitSchedulerService.getVisitById(visitId)
+
+  @PreAuthorize("hasRole('ROLE_VISIT_SCHEDULER')")
+  @GetMapping
+  @Operation(
+    summary = "Get visits",
+    description = "Retrieve visits with optional filters",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Visit Information Returned"
+      ),
+      ApiResponse(
+        responseCode = "400",
```

Comment:
I wonder if i is possible to get a 400 error ? Would in be if you didn't supply any filters? Which actually might be OK since it would just return all visits I guess

Reply:
yes - haven't decided on validation (if any). Maybe if the date is in the wrong format?